wrappers/python/tests/model/test_llama.py

```diff
@@ -167,73 +184,61 @@ def generate_inputs(params: LlamaTestParams):
         nntile_model.unregister()
 
         assert_close_by_frobnorm(
             torch_model.embed_tokens.weight.detach().numpy(),
             torch_model_other.embed_tokens.weight.detach().numpy(),
-            **dtype2tol[params.dtype]
+            **dtype2tol[dtype]
         )
-        for i in range(params.num_hidden_layers):
-            assert_close_by_frobnorm(
-                torch_model.layers[i].mlp.down_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].mlp.down_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-            assert_close_by_frobnorm(
-                torch_model.layers[i].mlp.up_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].mlp.up_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-            assert_close_by_frobnorm(
-                torch_model.layers[i].mlp.gate_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].mlp.gate_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-
-            assert_close_by_frobnorm(
-                torch_model.layers[i].post_attention_layernorm.weight.detach().numpy(),
-                torch_model_other.layers[i].post_attention_layernorm.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-            assert_close_by_frobnorm(
-                torch_model.layers[i].input_layernorm.weight.detach().numpy(),
-                torch_model_other.layers[i].input_layernorm.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-
-            assert_close_by_frobnorm(
-                torch_model.layers[i].self_attn.q_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].self_attn.q_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-            assert_close_by_frobnorm(
-                torch_model.layers[i].self_attn.v_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].self_attn.v_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-            assert_close_by_frobnorm(
-                torch_model.layers[i].self_attn.o_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].self_attn.o_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
-            assert_close_by_frobnorm(
-                torch_model.layers[i].self_attn.k_proj.weight.detach().numpy(),
-                torch_model_other.layers[i].self_attn.k_proj.weight.detach().numpy(),
-                **dtype2tol[params.dtype]
-            )
+        assert_close_by_frobnorm(
+            torch_model.norm.weight.detach().numpy(),
+            torch_model_other.norm.weight.detach().numpy(),
+            **dtype2tol[dtype]
+        )
+        for i in range(num_hidden_layers):
```

Comment:
Embed layer is checked, transformer blocks are checked, but the last rmsnorm is not checked fully. At least, bias `torch_model.norm.bias` is not checked here.

Reply:
Moreover, biases are not checked at all.

Reply:
The last norm is checked after the for-loop over the decoder blocks. RMSNorm has only a weight parameter.